libs/s25main/figures/nofWinegrower.cpp

```diff
@@ -0,0 +1,200 @@
+        unsigned draw_id;
+        if(now_id < 48)
+        {
+            draw_id = bobIndex[BobTypes::WINEGROWER_DIGGING_ANIMATION] + now_id % 15;
+
+            // Shovel-Sound
+            if(now_id % 8 == 3)
+            {
+                world->GetSoundMgr().playNOSound(76, *this, now_id, 200);
+                was_sounding = true;
+            }
+        } else
+            draw_id = bobIndex[BobTypes::WINEGROWER_PLANTING_ANIMATION] + now_id % 4;
+
+        LOADER.GetPlayerImage("wine_bobs", draw_id)->DrawFull(drawPt, COLOR_WHITE, world->GetPlayer(player).color);
+    }
+}
+
+unsigned short nofWinegrower::GetCarryID() const
+{
+    throw std::logic_error("Must not be called. Handled by custom DrawWalkingWithWare");
+}
+
+void nofWinegrower::WorkStarted()
+{
+    // Inform grapefield to not disappear during harvesting
+    if(harvest)
+        world->GetSpecObj<noGrapefield>(pos)->BeginHarvesting();
+}
+
+void nofWinegrower::WorkFinished()
+{
+    if(harvest)
+    {
+        // Destroy grapefield, and take before the ID of the harvested grapefield, which will be set as normal
+        // ornamental object
+        noBase* nob = world->GetNO(pos);
+        // Check if there is still a grapes field at this position
+        if(nob->GetGOT() != GO_Type::Grapefield)
+            return;
+        const unsigned wine_bobs = static_cast<noGrapefield*>(nob)->GetHarvestID();
+        world->DestroyNO(pos);
+        world->SetNO(pos, new noEnvObject(pos, wine_bobs, 7));
+
+        // Take harvested grapes into hands
+        ware = GoodType::Grapes;
+    } else
+    {
+        // If the point got bad (e.g. something was build), abort work
+        if(GetPointQuality(pos) == PointQuality::NotPossible)
+            return;
+
+        // What is currently here
+        const NodalObjectType noType = world->GetNO(pos)->GetType();
+
+        // Only demolish ornamental objects and signs
+        if(noType == NodalObjectType::Environment || noType == NodalObjectType::Nothing)
+        {
+            world->DestroyNO(pos, false);
+            // place new grapefield
+            world->SetNO(pos, new noGrapefield(pos));
+        }
+
+        // We have only planted (take nothing into hand)
+        ware = boost::none;
+    }
+
+    // recompute BQ around
+    world->RecalcBQAroundPoint(pos);
+}
+
+nofFarmhand::PointQuality nofWinegrower::GetPointQuality(const MapPoint pt) const
+{
+    // Either a grapefield exists we can harvest...
+    if(world->GetNO(pt)->GetType() == NodalObjectType::Grapefield)
+    {
+        if(world->GetSpecObj<noGrapefield>(pt)->IsHarvestable())
+            return PointQuality::Class1;
+        else
+            return PointQuality::NotPossible;
+    } else // or a free space, to place a new one
+    {
+        // Try to "plant" a new grapefield
+        // Still enough wares when starting new work (state = Waiting1)?
+        if(!workplace->WaresAvailable() && state == State::Waiting1)
```

Comment:
Use the "cheap" condition first:
```suggestion
        if(state == State::Waiting1 && !workplace->WaresAvailable())
```